lib/rex/socket/ssl.rb

```diff
@@ -122,7 +125,14 @@ def makessl(params)
       key, cert, chain = ssl_generate_certificate(cert_vars: {cn: params.ssl_cn})
     end
 
-    ctx = OpenSSL::SSL::SSLContext.new()
+    version = params&.ssl_version || DEFAULT_SSL_VERSION
+    # Raise an error if no selected versions are supported
+    unless Rex::Socket::SslTcp.system_ssl_methods.include? version
+      raise ArgumentError,
+        "This version of Ruby does not support the requested SSL/TLS version #{ssl_version}"
```

Comment:
This should really be `version` not `ssl_version`. Will fix this now.

```suggestion
        "This version of Ruby does not support the requested SSL/TLS version #{version}"
```

Reply:
Fixed in https://github.com/rapid7/rex-socket/pull/37/commits/a0de3ed28351db9720d7c842b7d7f67c6db5c7df